website/blog/2024-07-08-building-your-semantic-layer-in-pieces.md

```diff
@@ -9,90 +9,176 @@ date: 2024-07-10
+
+What this means for an analytics engineer working using dedicated SQL pools is that for every table model, it must have an `index` and `distribution` configured. In `dbt-synapse` the defaults are:
+
+- index: `CLUSTERED COLUMNSTORE INDEX`
+- distribution `ROUND_ROBIN`
+
+If you want something different, you can define it like below. For more information, see [dbt docs: configurations for Azure Synapse DWH: Indices and distributions](https://docs.getdbt.com/reference/resource-configs/azuresynapse-configs#indices-and-distributions).
+
+```sql
+{{
+    config(
+        index='HEAP',
+        dist='ROUND_ROBIN'
+        )
+}}
+SELECT * FROM {{ ref('some_model') }}
+```
+
+A distribution specifies how the table rows should be stored across the 60 nodes of the cluster. The goal is to provide a configuration that both:
+
+1. ensures data is split evenly across the nodes of the cluster, and
+2. minimizes inter-node movement of data.
+
+For example, imagine querying a 100-row seed table in a downstream model. Using `distribution=ROUND_ROBIN` instructs the pool to evenly distribute the rows between the 60 node, which equates to  having only one or two rows in each node. This `SELECT`-ing all these an operation that touches all 60 nodes. The end result is that the query will run much slower than you might expect.
+
+The optimal distribution is `REPLICATE` which will load a full copy of the table to every node. In this scenario, any node can return the 100 rows without coordination from the others. This is ideal for a lookup table which could limit the result set within each node before aggregating each nodes results.
+
+
+#### more information
+
+- [Guidance for designing distributed tables using dedicated SQL pool in Azure Synapse Analytics](https://learn.microsoft.com/en-us/azure/synapse-analytics/sql-data-warehouse/sql-data-warehouse-tables-distribute)
+- [source code for `synapse__create_table_as()` macro](https://github.com/microsoft/dbt-synapse/blob/master/dbt/include/synapse/macros/materializations/models/table/create_table_as.sql)
+
+
+## Deployments & Ecosystem
+
+With the infrastructure in place and the analytics engineers enabled with best practices, the final piece is to think through how a dbt project sits in the larger data stack of your organization both upstream and downstream.
+
+### Upstream
+
+In dbt, we assume the data has already been ingested into the warehouse raw. This follows a broader paradigm known as Extract-Load-Transform (ELT). The same goes for dbt with Azure Synapse. The goal should be to have the data ingested into Synapse that is as “untouched” as possible from when it came from the upstream source system. It’s common for data teams using Azure Data Factory to continue to imploy an ETL-paradigm where data is transformed before it even lands in the warehouse. We do not recommend this, as it results in critical data transformation living outside of the dbt project, and therefore undocumented.
```

Comment:
i think "ELT" and "ETL" should be in our custom dictionary (unless Vale starts ignoring acronyms)